data.rs

```diff
@@ -0,0 +1,170 @@
+use crate::PSP22Error;
+use ink::{
+    prelude::{vec, vec::Vec},
+    primitives::AccountId,
+    storage::Mapping,
+};
+
+pub enum PSP22Event {
+    Transfer(Option<AccountId>, Option<AccountId>, u128),
+    Approval(AccountId, AccountId, u128),
+}
+
+#[ink::storage_item]
+#[derive(Debug, Default)]
+pub struct PSP22Data {
+    total_supply: u128,
```

Comment:
maybe not in this PR, but I think we should be generic over balances type

Reply:
Again, the standard specifies it to be strictly `u128`.

Reply:
Well, to be precise, the standard defines a type alias `type Balance = u128;` which introduces name collision with the ink env type for native chain currency. Hence I decided to drop that alias.